File: models/Addiction.js
import mongoose from "mongoose";

const AddictionSchema = new mongoose.Schema({
    patientId: { type: mongoose.Schema.Types.ObjectId, ref: "PatientRegistration" },
    type: String, // Tobacco, Alcohol
    duration: String,
    units: Number,
    frequency: String,
    createdByUser:{type:mongoose.Schema.Types.ObjectId,ref:"User"},
    status: { type: String, enum: ["Ongoing", "Stopped"] }
  }, { timestamps: true });
  
  export default mongoose.model("Addiction", AddictionSchema);
  

File: jstojv/src/main/java/com/samrat/jstojv/entity/Addiction.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "addictions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Addiction extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id")
    private PatientRegistration patient;

    private String type; // Tobacco, Alcohol
    private String duration;
    private Integer units;
    private String frequency;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdByUser;

    @Enumerated(EnumType.STRING)
    private Status status;

    public enum Status {
        Ongoing, Stopped
    }
}
